lib/services/header.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package services
+
+import (
+	"bytes"
+	"sync"
+
+	"github.com/gogo/protobuf/jsonpb"
```

Comment:
Aren't we trying to deprecate the usage of gogo?

Reply:
Yes, but I don't know how doable that will be in this scenario. The `Plugin` object sits in `types.proto`, so in order to prevent using gogo here seems like it would be kind of a nasty dance.